src/SFA.DAS.AssessorService.Web/Controllers/Apply/ApplicationController.cs

```diff
@@ -523,23 +523,25 @@ public async Task<IActionResult> SaveMultiplePageAnswers(Guid Id, int sequenceNo
                             }
                             return RedirectToAction("Feedback", new { Id });
                         }
-                        else if (pageAddResponse.ValidationErrors?.Count == 0 || answers.All(x => x.Value == string.Empty || Regex.IsMatch(x.Value, "^[,]+$")))
+                        else if (( pageAddResponse != null && pageAddResponse.ValidationErrors?.Count == 0) || answers.All(x => x.Value == string.Empty || Regex.IsMatch(x.Value, "^[,]+$")))
                         {
                             var nextAction = page.Next.SingleOrDefault(x => x.Action == "NextPage");
 
-                            if (!string.IsNullOrEmpty(nextAction.Action))
+                            if (!string.IsNullOrEmpty(nextAction?.Action))
                             {
                                 if (__redirectAction == "Feedback")
                                 {
+                                    var redirectNextAction = false;
                                     foreach (var answer in answers)
                                     {
                                         if (page.Next.Exists(y => y.Conditions.Exists(x => x.QuestionId == answer.QuestionId || x.QuestionTag == answer.QuestionId)))
                                         {
-                                            return RedirectToNextAction(Id, sequenceNo, sectionNo, nextAction.Action, nextAction.ReturnId, __redirectAction, "Hide");
+                                            redirectNextAction = true;
```

Comment:
The reason for this change was to remove the 'break;'. This is shown on the next line (540). SonarCloud reported the 'break;' as not allowing a foreach loop to iterate more than once. Please see the SonarCloud bug [here](https://sonarcloud.io/project/issues?resolved=false&severities=MAJOR&types=BUG&id=SkillsFundingAgency_das-assessor-service&open=AXGndfR8JQ_u_BPQDvVz).

Reply:
The original code took the first answer and then called RedirectToNextAction if that answer met the if condition, the new code will compare each answer and if any of them meet the if condition then it will RedirectToNextAction.

To rewrite this without using the break (will have to assume the code was correct to start with!) then need to take just the first answer i.e  
var answer = answers.FirstOrDefault()
if(answer != null && page.Next.Exists.... ) return RedirectToNextAction